packages/web-components/src/message-bar/message-bar.ts

```diff
@@ -0,0 +1,70 @@
+import { attr, FASTElement } from '@microsoft/fast-element';
+import { MessageBarIntent, MessageBarLayout, MessageBarPoliteness, MessageBarShape } from './message-bar.options.js';
+
+/**
+ * A Message Bar Custom HTML Element.
+ *
+ * @slot actions - Content that can be provided for the actions
+ * @slot close - Content that can be provided for the close button
+ * @slot - The default slot for the content
+ * @public
+ */
+export class MessageBar extends FASTElement {
+  /**
+   * Sets the aria-labelledby attribute of the control.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `aria-labelledby`
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  ariaLabelledBy: string | null = null;
+
+  /**
+   * Sets the shape of the control.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `shape`
+   */
+  @attr
+  shape: MessageBarShape = 'rounded';
+
+  /**
+   * Sets the layout of the control.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `layout`
+   */
+  @attr
+  layout: MessageBarLayout = 'singleline';
+
+  /**
+   * Sets the intent of the control.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `intent`
+   */
+  @attr
+  intent: MessageBarIntent = 'info';
```

Comment:
I'm going to be pushing a change to button/anchor button to use custom states with an attribute fallback - I think this is going to be "the way" given performance investigations showing no delta and in some cases, milliseconds of gain. Let's see if we can use those patterns here. I'll link the PR when it's up.